### PR TITLE
[PUB-3423] Add campaigns paywall

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
- Add campaigns paywall: users without campaigns will see the Empty Page with a CTA to upgrade their plan
- Tracking for the free plan paywall action
- While working on it I realized there's no mention to campaigns in the plans page nor the [pricing page](https://buffer.com/pricing/publish#compare-features). After talking to Lauren I decided to add it to our plans page.

## Context & Notes
https://buffer.atlassian.net/browse/PUB-3423

## Screenshots (if appropriate)
Screenshot 2021-03-15 at 18 53 45
Screenshot 2021-03-15 at 18 53 55

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/campaigns-list/index.js [hunk 1]
```diff
@@ -45,6 +47,14 @@ export default connect(
         );
       }
     },
+    onUpgradeButtonClick: () => {
+      dispatch(
+        analyticsActions.trackEvent('Upgrade Path Viewed', {
+          upgradePathName: 'campaigns-emptyState-upgrade',
```
Comment (anafilipadealmeida) [inline packages/campaigns-list/index.js:53]: I saw Hamish stopped using the SEGMENT_NAMES constants for this. I'm wondering, are we still doing it?

Comment (albennett) [reply packages/campaigns-list/index.js:53]: Hey! Yes, Hamish added the tracking changes after I had reviewed it, so I didn't catch it. I did reach out to him about adding it to the config file, so we can keep the segment names in one place. So yes, we're still doing it.

File: packages/campaigns-list/components/ListCampaigns/index.jsx [hunk 2]
```diff
@@ -56,21 +55,22 @@ const ListCampaigns = ({
...
-    window.location = getURL.getPublishUrl();
-    return null;
-  }
-
   const { t } = useTranslation();
 
-  if (!isLoading && (!campaigns || campaigns.length === 0)) {
+  if (
+    (!isLoading && (!campaigns || campaigns.length === 0)) ||
+    !hasCampaignsFlip
```
Comment (albennett) [inline packages/campaigns-list/components/ListCampaigns/index.jsx:64]: We should have made these changes before the org switcher work 😂

File: packages/campaigns-list/components/ListCampaigns/EmptyState/index.jsx [hunk 3]
```diff
@@ -65,8 +67,16 @@ const EmptyState = ({
...
-                label={t('campaigns.common.createCampaign')}
-                onClick={onOpenCreateCampaignClick}
+                label={
+                  shouldDisplayPaywall
+                    ? t('campaigns.common.upgradePlan')
+                    : t('campaigns.common.createCampaign')
+                }
+                onClick={
+                  shouldDisplayPaywall
+                    ? onUpgradeButtonClick
```
Comment (albennett) [inline packages/campaigns-list/components/ListCampaigns/EmptyState/index.jsx:77]: I lean towards trying to add a test if possible with anything upgrade-related. I wonder if a behavioral test could make sense here?

**Review by albennett — APPROVED** [review]
Thanks Ana, looks great 🚀

**Pull request merged**